transformer_engine/common/include/transformer_engine/cast_transpose_noop.h

```diff
@@ -0,0 +1,35 @@
+/*************************************************************************
+ * Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+
+/*! \file transpose_with_noop.h
+ *  \brief Functions handling transposes with no-op.
+ */
+
+#ifndef TRANSFORMER_ENGINE_CAST_TRANSPOSE_WITH_NOOP_H_
+#define TRANSFORMER_ENGINE_CAST_TRANSPOSE_WITH_NOOP_H_
+
+#include "transformer_engine.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+void nvte_transpose_with_noop(const NVTETensor input,
+                              const NVTETensor noop,
```

Comment:
We're treating `noop` as a boolean flag, so the intent would be clearer with:
```suggestion
                              const bool *noop,
```

Reply:
The tensor is needed for cuda graph capture

Reply:
In an ideal world the dtype would be bool, but getting this right may not be worth the delay.